### PR TITLE
Merge 0.66-stable into main

### PR DESCRIPTION
**Pull request by amgleitman**
## Summary

Move changes made to microsoft/0.66-stable into the main branch.

These include bug fixes as well as improvements to our CI checks.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -84,7 +84,7 @@ We use GitHub issues to track bugs exclusively. We have documented our issue han
 
 ### Merging upstream master into your fork master
 
-From time to time, your fork will get out of sync with the upstream remote. Use the following commands to get the master branch of your fork up up to date.
+We use GitHub issues to track bugs exclusively. We have documented our issue handling processes in the [Issues wiki](https://github.com/facebook/react-native/wiki/Triaging-GitHub-Issues).
```
Comment (HeyImChris) [inline CONTRIBUTING.md:87]: We're getting rid of our advice on how to sync with FB's upstream repo and replacing it with the upstream issue wiki? I'd imagine we at least want to link the rn-macos Issues template if we change this right?

File: scripts/update_podfile_lock.sh [hunk 2]
```diff
@@ -22,14 +22,16 @@ validate_env () {
...
 
-  COCOAPODS_VERSION=$(pod --version)
-  if [[ "$COCOAPODS_VERSION" != "$REQUIRED_COCOAPODS_VERSION" ]];
-  then
-    echo "You must have CocoaPods $REQUIRED_COCOAPODS_VERSION installed; you have $COCOAPODS_VERSION."
-    echo "Installing via gem is recommended:"
-    echo "  sudo gem install cocoapods -v $REQUIRED_COCOAPODS_VERSION"
-    exit 1
-  fi
+  # // [(macOS) Commenting this out to avoid hitting breaks everytime the agents images are updated
```
Comment (HeyImChris) [inline scripts/update_podfile_lock.sh:25]: Nit: should have a  TODO tag with a bug #

**Review by HeyImChris — APPROVED** [review]
Can you add the git commands you ran to merge 0.66-stable over to master? Just to make it super clear in the PR where this code is coming from

File: CONTRIBUTING.md (hunk 1 above)
Comment (amgleitman) [reply CONTRIBUTING.md:87]: This added line is just a copy of another line that already appears in this file. I'm assuming this was an accident? Either way, we should definitely revert it back.

**Pull request merged**